### ISSUE TITLE
`extract_sw_logs.py` sometimes fails while extracting strings

### ISSUE THREAD

**Issue by cfrantz** [lowRISC/opentitan#6644]
The tool `extract_sw_logs.py` occasionally fails while extracting strings.  I've observed this behavior while writing the HMAC functional test for the mask rom: I have a constant which is a sha256 digest:

```
const uint32_t kGettysburgDigest[] = {
    0x1e6fd403, 0x0f9034cd, 0x775708a3, 0x96c324ed,
    0x420ec587, 0xeb3dd433, 0xe29f6ac0, 0x8b8cc7ba,
};
```

When this constant is marked `static`, extract_sw_logs fails with this message:

```
FAILED: sw/device/silicon_creator/testing/sw_silicon_creator_lib_driver_hmac_functest_sim_dv.rodata.txt sw/device/silicon_creator/testing/sw_silicon_creator_lib_driver_hmac_functest_sim_dv.logs.txt 
/usr/bin/python3 /usr/local/google/home/cfrantz/opentitan/opentitan/util/device_sw_utils/extract_sw_logs.py --elf-file sw/device/silicon_creator/testing/sw_silicon_creator_lib_driver_hmac_functest_sim_dv.elf --rodata-sections .rodata --logs-fields-section .logs.fields --name sw_silicon_creator_lib_driver_hmac_functest_sim_dv --outdir sw/device/silicon_creator/testing
Traceback (most recent call last):
  File "/usr/local/google/home/cfrantz/opentitan/opentitan/util/device_sw_utils/extract_sw_logs.py", line 262, in <module>
    main()
  File "/usr/local/google/home/cfrantz/opentitan/opentitan/util/device_sw_utils/extract_sw_logs.py", line 249, in main
    rodata, result = extract_sw_logs(args.elf_file, args.logs_fields_section,
  File "/usr/local/google/home/cfrantz/opentitan/opentitan/util/device_sw_utils/extract_sw_logs.py", line 174, in extract_sw_logs
    addr_strings = get_addr_strings(ro_contents)
  File "/usr/local/google/home/cfrantz/opentitan/opentitan/util/device_sw_utils/extract_sw_logs.py", line 136, in get_addr_strings
    raise IndexError(exc_msg)
IndexError: Error: duplicate {addr: string} pair encountered
addr: 536876452 string: �?
addr: 536876452 string: �?
```

**Comment by alphan** [comment]
#6647 should fix this. @sriyerg can you take a look?

**Comment by imphil** [comment]
Fixed by https://github.com/lowRISC/opentitan/pull/6647.

(@alphan: can you add a `Fixes #6644` to the commit message next time to link the issue with the change?)
